amr-wind/ocean_waves/relaxation_zones/waves2amr_ops.H

```diff
@@ -785,9 +804,18 @@ struct UpdateTargetFieldsOp<W2AWaves>
             // Only perform reading where needed, communicate offset though
             amrex::ParallelDescriptor::ReduceIntMax(wdata.n_offset);
 
-            // If double read is required, then copy older wave data to ow_
-            // fields and modify interpolation parameters to get things right
-            if (double_data == 1) {
+            // Previous W2A data is more recent than last interpolated data
+            t_last = (wdata.ntime - 1) * wdata.dt_modes;
```

Comment:
this line, and the copying under double_data = 0, are the key changes to improve the implementation

Reply:
This is great. Nice job!